baybe/constraints/continuous.py

```diff
@@ -31,3 +36,87 @@ class ContinuousLinearInequalityConstraint(ContinuousConstraint):
     The class has no real content as it only serves the purpose of
     distinguishing the constraints.
     """
+
+
+@define
+class ContinuousCardinalityConstraint(ContinuousConstraint):
+    """Class for continuous cardinality constraints.
+
+    Places a constraint on the set of nonzero (i.e. "active") values among the
+    specified parameter, bounding it between the two given integers,
+        ``min_cardinality`` <= | {1(p_i != 0)}_i | <= ``max_cardinality``
+    where ``1`` is the Kronecker delta function and ``{p_i}`` are the parameters
+    specified for the constraint.
+
+    Note that this can be equivalently regarded a L0-constraint on the vector containing
+    the specified parameters.
+    """
+
+    min_cardinality: int = field(default=0, validator=[instance_of(int), ge(0)])
+    "The minimum required cardinality."
+
+    max_cardinality: int = field(validator=instance_of(int))
+    "The maximum allowed cardinality."
+
+    @max_cardinality.default
+    def _default_max_cardinality(self):
+        """Use the number of involved parameters as the upper limit by default."""
+        return len(self.parameters)
+
+    def __attrs_post_init__(self):
+        """Validate the cardinality bounds."""
+        if self.min_cardinality > self.max_cardinality:
+            raise ValueError(
+                f"The lower cardinality bound cannot be larger than the upper bound. "
+                f"Provided values: {self.max_cardinality=}, {self.min_cardinality=}."
+            )
+
+        if self.max_cardinality > len(self.parameters):
+            raise ValueError(
+                f"The cardinality bound cannot exceed the number of parameters. "
+                f"Provided values: {self.max_cardinality=}, {len(self.parameters)=}."
+            )
+
+        if self.min_cardinality == 0 and self.max_cardinality == len(self.parameters):
+            raise ValueError(
+                f"No constraint of type `{self.__class__.__name__}' is required "
+                f"when 0 <= cardinality <= len(parameters)."
+            )
+
+    def sample_inactive_parameters(self, batch_size: int = 1) -> list[set[str]]:
```

Comment:
we use `batch_size` in a very important specific context, namely the number of recommendations a user is getting
Can we choose a different name here where its snot even about datapoints but parameters?

Reply:
as agreed, we'll keep `batch_size` in all searchspace-related contexts